bench/workgen/workgen.cxx

```diff
@@ -248,6 +281,231 @@ WorkloadRunner::start_table_idle_cycle(WT_CONNECTION *conn)
+                    << _workload->options.dynamic_create_target << " MB.");
+            }
+        } else {
+            creating = db_size < _workload->options.dynamic_create_trigger;
+            if (creating) {
+                VERBOSE(*_workload,
+                  "Started creating new tables. db_size now "
+                    << db_size << " MB has reached the create trigger of "
+                    << _workload->options.dynamic_create_trigger << " MB.");
+            }
+        }
+        if (!creating) {
+            sleep(_workload->options.dynamic_create_period);
+            continue;
+        }
+
+        for (int creates = 0; creates < _workload->options.dynamic_create_count;
+             ++creates, ++total_creates) {
+            snprintf(uri, BUF_SIZE, "table:test_dynamic_%" PRIu64, total_creates);
+
+            // Make sure that this table is not a part of static table set, continue if it is.
+            if (icontext->_tint.count(uri) > 0 || icontext->_dyn_tint.count(uri) > 0)
+                continue;
+
+            // Create the table.
+            if ((ret = session->create(session, uri, "key_format=S,value_format=S")) != 0) {
+                if (ret == EBUSY)
+                    continue;
+                THROW("Table create failed in start_dynamic_table_mgmt.");
+            }
+
+            // The dynamic table data structures are protected by a mutex.
+            {
+                const std::lock_guard<std::mutex> lock(*icontext->_dyn_mutex);
+
+                // Add the table into the list of dynamic set.
+                tint_t tint = icontext->_dyn_tint_last;
+                icontext->_dyn_tint[uri] = tint;
+                icontext->_dyn_table_names[tint] = uri;
+                icontext->_dyn_table_runtime[tint] = TableRuntime();
+                ++icontext->_dyn_tint_last;
+                VERBOSE(*_workload, "Created table and added to the dynamic set: " << uri);
+            }
+        }
+
+        sleep(_workload->options.dynamic_create_period);
+    }
+
+    return 0;
+}
+
+/*
+ * This function will drop "dynamic_drop_count" tables every "dynamic_drop_period" seconds. It will
+ * also monitor the database size and stop dropping tables once the size goes below
+ * "dynamic_drop_target". It will restart table drops if the database size crosses
+ * "dynamic_drop_trigger".
+ */
+int
+WorkloadRunner::start_dynamic_drop(WT_CONNECTION *conn)
+{
+    WT_SESSION *session;
+    if (conn->open_session(conn, nullptr, nullptr, &session) != 0) {
+        THROW("Error opening a session.");
+    }
+
+    WT_DECL_RET;
+    ContextInternal *icontext = _workload->_context->_internal;
+    std::vector<std::string> tables_delete; // Track tables that are to be deleted
+    uint32_t db_size = get_dir_size_mb(_wt_home);
+
+    // Initially we start dropping tables if the database size is more than the drop trigger.
+    bool dropping = db_size > _workload->options.dynamic_drop_trigger;
```

Comment:
I would have the same suggestion here, I don't think we need `dropping` to be defined here. It could be defined inside the `while` loop`.

Reply:
Look at the comment before, this is intentional.